src/Input.test.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+import { render, fireEvent } from '@testing-library/react';
+
+import Input from './Input';
+
+describe('<Input /> ', () => {
+  context('rendering 되면', () => {
+    it('input과 button으로 이루어져 있는지 확인한다.', () => {
+      const { getByText, getByPlaceholderText } = render(<Input />);
+      expect(getByPlaceholderText('할 일을 입력해 주세요')).toHaveAttribute('type', 'text');
+      expect(getByText('추가')).toHaveAttribute('type', 'button');
+    });
+  });
+
+  context('change 이벤트가 발생하면', () => {
+    it('value의 값이 바뀐다.', () => {
+      const { getByPlaceholderText } = render(<Input />);
+      const input = getByPlaceholderText('할 일을 입력해 주세요');
+      fireEvent.change(input, {
+        target: {
+          value: '자바스크립트 공부하기',
+        },
+      });
+      expect(input.value).toBe('자바스크립트 공부하기');
+    });
```

Comment:
지금은 `value` attribute에 값을 안넣어줘서 그런데 빈 값이라도 넣어주면 이 테스트 통과하나요?

Reply:
`<Input />` 컴포넌트의 value 말씀하시는거죠 ?

Reply:
네넵

Reply:
통과하지 않네요; value의 기본 값을 빈 값을 넣어주고 change 이벤트를 발생시킨건데 왜 변하지 않는걸까요..?

Reply:
`Input`컴포넌트의 `<input />` value는 props의 value에 의해서만 변경되기 때문이에요

Reply:
근데 생각해보니 Input 컴포넌트에서 change 이벤트를 테스트 하는게 맞는걸까요? change 이벤트의 기능은 App에 구현되어 있으니 App에서 하는게 맞지 않을까요 ?

Reply:
어떤 테스트를 하고 싶은지 맥락이 중요할 것 같아요 ㅎㅎ 저는 App에서 해주는게 맞다고 생각했어요. 정답이 있는건 아닙니다

Reply:
> `Input`컴포넌트의 `<input />` value는 props의 value에 의해서만 변경되기 때문이에요

무슨 뜻인지 잘 모르겠어요. value에 빈 값을 전달했으면 그게 props로 value를 전달한 것 아닌가요? 빈 값을 넣어주었을 때 뜨는 에러가 다음과 같았습니다.

> failed prop type: you provided a `value` prop to a form field without an `onchange` handler. this will render a read-only field. if the field should be mutable use `defaultvalue`. otherwise, set either `onchange` or `readonly`

변경 핸들러 없이 value를 전달해서 실패했다고 해서 `jest.fn()` 을 통해 mocking 함수를 만들어 전달했더니 통과했어요.